### PR TITLE
Add SST-2 NLP example.

### PR DESCRIPTION
**Pull request by bastings**
This adds an NLP example: text classification on the SST-2 data set.
The model is a simple LSTM with MLP classifier on top, trained with binary cross entropy loss.

### CODE REVIEW

File: examples/sst2/input_pipeline.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+from typing import Dict, Sequence, Text
+
+from absl import logging
+
+import numpy as np
+import tensorflow.compat.v2 as tf
+import tensorflow_datasets as tfds
+
+def build_vocab(datasets: Sequence[tf.data.Dataset],
+                special_tokens: Sequence[Text] = (b'<pad>', b'<unk>', b'<s>', b'</s>'),
```
Comment (bohnetbd) [inline examples/sst2/input_pipeline.py]: I was wondering why to use bytes and not python strings?

File: examples/sst2/model.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  model_def = TextClassifier.partial(train=False, **model_kwargs)
+  _, initial_params = model_def.init_by_shape(
+      jax.random.PRNGKey(seed),
+      [((batch_size, max_len), jnp.int32),
+       ((batch_size,), jnp.int32)])
+  model = nn.Model(model_def, initial_params)
+  return model
+
+
+def word_dropout(inputs: jnp.ndarray, rate: float = None, unk_idx: int = None):
```
Comment (bohnetbd) [inline examples/sst2/model.py]: Very nice functionality! I was wondering about the interface if this could be more similar to other modules, providing train, and not having default None values for rate and unk_idx. Maybe, just adapting train. Up to you of cause.

File: examples/sst2/model.py [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+  def apply(self,
+            inputs: jnp.ndarray,
+            hidden_size: int = None,
+            output_size: int = None,
+            output_bias: bool = False,
+            dropout: float = None,
+            train: bool = None):
+    # inputs.shape = <float32>[batch_size, seq_length, hidden_size]
+    hidden = nn.Dense(inputs, hidden_size, name='hidden')
+    hidden = nn.tanh(hidden)
```
Comment (bohnetbd) [inline examples/sst2/model.py]: No action needed but relu or elu lead often to better results and more numerical stability - no nan.

File: examples/sst2/input_pipeline.py (hunk 1 above)
Comment (bastings) [reply examples/sst2/input_pipeline.py]: TFDS is returning text fields as bytes so I sticked to that for performance. Sounds like a good idea to make this consistent where possible.

File: examples/sst2/train.py [hunk 4]
```diff
@@ -0,0 +1,352 @@
...
+    eval_ds: A dataset to be used for the evaluation. Typically valid or test.
+
+  Returns:
+    A dict with the evaluation results.
+  """
+  count = 0
+  total_loss = 0.
+  total_correct = 0
+
+  for ex in tfds.as_numpy(eval_ds):
```
Comment (bohnetbd) [inline examples/sst2/train.py]: Would not pass google python review :) No abbreviations that are not standard.

Comment (bastings) [reply examples/sst2/train.py]: Fixed.

File: examples/sst2/model.py (hunk 2 above)
Comment (bastings) [reply examples/sst2/model.py]: I added "deterministic" like dropout has, so it's now a more standard API.

File: examples/sst2/model.py (hunk 3 above)
Comment (bastings) [reply examples/sst2/model.py]: I'll experiment with it and update if it indeed works better.

**Comment by avital** [comment]
Could you please also add a link from the README to this example?
